src/bulk_insert/bulk_insert.c

```diff
@@ -86,6 +90,15 @@ static inline SIValue _BulkInsert_ReadProperty(const char *data, size_t *data_id
 		*data_idx += strlen(s) + 1;
 		// The string itself will be cloned when added to the GraphEntity properties.
 		v = SI_ConstStringVal((char *)s);
+	} else if(t == BI_ARRAY) {
+		// The first 8 bytes of a received array will be the array length.
+		int64_t len = *(int64_t *)&data[*data_idx];
```

Comment:
How about
```
int64_t len;
memcpy(&len, data[*data_idx], sizeof(len));
```

Reply:
I can change this if you like, but this is the convention used throughout this function. I can change all of them if you prefer?